### PR TITLE
fix: new arch android interface

### LINKED ISSUE #652: slider has 0 height on Android with new architecture enabled
## Description

While testing the project (I tested from the main branch) with new architecture (RN 0.75) I experienced 2 issues:

1. project does not build following https://github.com/callstack/react-native-slider/pull/649 because the codegen-generated interface expects `float`s instead of `double`s - that's easy to fix
2. slider cannot be interacted with (dragged) because it has zero height; see image

Screenshot 2024-10-05 at 0 00 02

This seems like a regression since https://github.com/callstack/react-native-slider/pull/589 which was solving the same problem.
I'd be happy to submit a PR instead of an issue but I believe you'll be much better at figuring it out :).

## Reproducible Demo

Please see the example at https://github.com/vonovak/react-native-slider/tree/fix/new-arch-issues/new-example

It's a fork of this repo, with new example added which uses https://github.com/microsoft/react-native-test-app

The example app was generated with `npx --package react-native-test-app@latest init` and then I modified `metro.config.js` and `react-native.config.js` to make it all work. Then I copied src directory from the existing example.

### PR DESCRIPTION
**Pull request by draggie**
Summary:
---------

fixes #652 
Resolved incompatible interface generation for the codegen source.

Test Plan:
----------

### CODE REVIEW

**Review by vonovak — COMMENTED** [review]
I didn't run the code but it looks okay to me, just added a comment about using doubles.

re "Tmp patch for invalid height calculation" in the description: that doesn't seem to be included

**Comment by draggie** [comment]
@vonovak Thanks, I have removed the tmp solution from the description as after further investigation this issue is related to `measure` function from `ViewManager` not being called in some setups like RNTA

File: package/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java [hunk 1]
```diff
@@ -125,12 +125,12 @@ public void setValue(ReactSlider view, float value) {
...
   }
 
-  @ReactProp(name = "minimumValue", defaultFloat = 0f)
+  @ReactProp(name = "minimumValue", defaultDouble = 0d)
   public void setMinimumValue(ReactSlider view, double value) {
     ReactSliderManagerImpl.setMinimumValue(view, value);
   }
 
-  @ReactProp(name = "maximumValue", defaultFloat = 1f)
+  @ReactProp(name = "maximumValue", defaultDouble = 1f)
```
Comment (vonovak) [inline package/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java:133]:
```suggestion
  @ReactProp(name = "maximumValue", defaultDouble = 1d)
```

File: package/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java [hunk 2]
```diff
@@ -145,8 +145,8 @@ public void setUpperLimit(ReactSlider view, float value) {
     ReactSliderManagerImpl.setUpperLimit(view, value);
   }
 
-  @ReactProp(name = "step", defaultFloat = 0f)
-  public void setStep(ReactSlider view, float value) {
+  @ReactProp(name = "step", defaultDouble = 0f)
```
Comment (vonovak) [inline package/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java:148]:
```suggestion
  @ReactProp(name = "step", defaultDouble = 0d)
```

**Review by vonovak — CHANGES_REQUESTED** [review]

**Comment by chrfalch** [comment]
Me and @vonovak made #657 - which might solve the root cause of the problem.

**Review by okwasniewski — APPROVED** [review]

**Review by BartoszKlonowski — APPROVED** [review]

**Pull request merged**